File: Tree/Postorder.py
class Node:
    def __init__(self, data, left=None, right=None):
        self.data = data
        self.left = left
        self.right = right

    def postorder(self, root):
        if root is None:
            return
        self.postorder(root.left)
        self.postorder(root.right)
        print(root.data, end = ' ')
        


    def main(self):
        root = Node(1)
        root.left = Node(2)
        root.right = Node(3)

        self.postorder(root)

if __name__ == "__main__":
    obj = Node(0)
    obj.main()

File: Tree/Postorder.java
class Node{
    int data;
    Node left, right;

    Node(int data){
        this.data = data;
        left=right=null;
    }
}


public class Postorder {

    static void postorder_traversal(Node root){
        if(root==null) return;

        postorder_traversal(root.left);
        postorder_traversal(root.right);
        System.out.print(root.data+" ");
       
    }

    public static void main(String[] args) {
        Node root = new Node(1);
        root.left = new Node(2);
        root.right = new Node(3);

        postorder_traversal(root);
    }
    
}
